webpack.config.js

```diff
@@ -1,90 +1,31 @@
+/**
+ * External dependencies
+ */
+const _ = require( 'lodash' ); // eslint-disable-line lodash/import-scope
 const path = require( 'path' );
-const webpack = require( 'webpack' );
-const MiniCssExtractPlugin = require( 'mini-css-extract-plugin' );
-const WordPressExternalDependenciesPlugin = require( '@automattic/wordpress-external-dependencies-plugin' );
+const getBaseWebpackConfig = require( '@automattic/calypso-build/webpack.config.js' );
 
-const NODE_ENV = process.env.NODE_ENV || 'development';
-const devMode = process.env.NODE_ENV !== 'production';
+const isDevelopment = process.env.NODE_ENV !== 'production';
 
-const webpackConfig = {
-	mode: devMode ? 'development' : 'production',
-	// Entry points point to the javascript module
-	// that is used to generate the script file.
-	// The key is used as the name of the script.
-	entry: {
-		admin: path.join( __dirname, './_inc/client/admin.js' ),
-		static: path.join( __dirname, './_inc/client/static.jsx' ),
-	},
-	output: {
-		path: path.join( __dirname, '_inc/build' ),
-		filename: '[name].js',
-	},
-	module: {
-		// Webpack loaders are applied when a resource is matches the test case
-		rules: [
-			{
-				test: /\.jsx?$/,
-				loader: 'babel-loader',
-				// include both typical npm-linked locations and default module
-				// locations to handle both cases
-				include: [ path.join( __dirname, 'test' ), path.join( __dirname, '_inc/client' ) ],
-			},
-			{
-				test: /\.css$/,
-				use: [ { loader: MiniCssExtractPlugin.loader }, 'css-loader', 'autoprefixer-loader' ],
-			},
-			{
-				test: /\.html$/,
-				loader: 'html-loader',
-			},
-			{
-				test: /\.scss$/,
-				use: [ { loader: MiniCssExtractPlugin.loader }, 'css-loader', 'sass-loader' ],
-			},
-			{
-				test: /\.svg/,
-				loader: 'url-loader',
-			},
-		],
-	},
+const webpackConfig = getBaseWebpackConfig(
+	{ WP: true },
+	{
+		entry: {
+			admin: path.join( __dirname, '_inc', 'client', 'admin.js' ),
+			static: path.join( __dirname, '_inc', 'client', 'static.jsx' ),
+		},
+		'output-filename': '[name].js',
+		'output-path': path.join( __dirname, '_inc', 'build' ),
+	}
+);
+
+module.exports = _.merge( {}, webpackConfig, {
```

Comment:
Shouldn't we expand with `...` like we do in `webpack.config.extensions.js` rather than use `_.merge`? I can't remember details now what got overwritten with `merge`, tho. Plugins?

Reply:
The issue isn't specific to plugins, but to `_.merge()`ing arrays (to extend them with additional entries), see #11967:
 
> Turns out `_.merge()` caused `CopyWebpackPlugin` to override the first item in `webpackConfig.plugins` ([`DefinePlugin`](https://github.com/Automattic/wp-calypso/blob/a75746a27bf87c546b34ca048bacd362a6736738/packages/calypso-build/webpack.config.js#L104=L107)) :roll_eyes: 

In this PR, we aren't trying to add additional entries to an existing array, so `_.merge()` should be fine :+1:

Reply:
Yup, also just double checked by console.logging that everything stays as expected. 👍 

I'd still feel more comfortable if there weren't a chance that someone might run into this issue by modifying this config but I don't feel strongly either way.

Reply:
That's a fair point, yup.